src/Raven.Client/Documents/Conventions/DocumentConventions.cs

```diff
@@ -837,6 +848,51 @@ public bool TryConvertValueForQuery(string fieldName, object value, bool forRang
             return false;
         }
 
+        private void RegisterQueryValueConverter<T>(TryConvertValueToObjectForQueryDelegate<T> converter)
+        {
+            AssertNotFrozen();
+
+            int index;
+            for (index = 0; index < _listOfQueryValueToObjectConverters.Count; index++)
+            {
+                var entry = _listOfQueryValueToObjectConverters[index];
+                if (entry.Type.IsAssignableFrom(typeof(T)))
+                    break;
+            }
+
+            _listOfQueryValueToObjectConverters.Insert(index, (typeof(T), Actual));
+
+            bool Actual(string name, object value, bool forRange, out object objValue)
+            {
+                if (value is T)
+                    return converter(name, (T)value, forRange, out objValue);
+                objValue = null;
+                return false;
+            }
+        }
+
+        public void RegisterQueryValueConverter<T>(TryConvertValueToObjectForQueryDelegate<T> converter, RangeType rangeType)
+        {
+            RegisterQueryValueConverter(converter);
+
+            if (_customRangeTypes.ContainsKey(typeof(T)) == false)
```

Comment:
This might be confusing to the user, if they try to register two value converters, they'll get the first one?

Reply:
Yes, so how I want to register a converter that can strongly typed convert to string (the existing API)
but I also want to register a version that can convert to object. (a new API).